### PR TITLE
fix: bg video no cover image

### PR DESCRIPTION
**Pull request by tanflem**
# Description

### <samp>🤖 Generated by Copilot at 8ada33c</samp>

Updated the logic for displaying poster images on video cards. Now, only videos that are not from YouTube or Cloudflare will use the poster image from the journey entry. This supports the new feature of uploading and managing videos from Cloudflare.

- [Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/33281894/todos/6384689751)

# How should this PR be QA Tested?
1. Upload a custom video as the background media.
2. Cover Image of the video should display.

# Walkthrough

### <samp>🤖 Generated by Copilot at 8ada33c</samp>

*  Exclude Cloudflare videos from using poster image on video component ([link](https://github.com/JesusFilm/core/pull/1803/files?diff=unified&w=0#diff-de466867478c6ebdafd8654eee49297476e8dd17f7c99bc8ed3b552c1e161d70L70-R71))

### CODE REVIEW

File: libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx [hunk 1]
```diff
@@ -67,7 +67,8 @@ export function ContainedCover({
   const contentRef = useRef()
 
   const posterImage =
-    videoBlock?.source !== VideoBlockSource.youTube
+    videoBlock?.source !== VideoBlockSource.youTube &&
+    videoBlock?.source !== VideoBlockSource.cloudflare
```
Comment (storyworks) [inline libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx:71]: What's the data structure for a cloudflare video? I thought `videoBlock.image` was only set on youtube videos - if this is the case, how would this change set a default image for cloudflare videos?

Comment (tanflem) [reply libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx:71]: The default image for Cloudflare videos was already being set. It was not being displayed for custom videos set as background media.

Here is the structure. The image is stored in video.image 

```
              {
                  "__typename": "VideoBlock",
                  "id": "92f5cd2e-3a0c-4aea-8a66-a5cf219f9d5d",
                  "parentBlockId": "fbbb1ec9-a923-4dbd-a1c9-01d88dd660bc",
                  "parentOrder": null,
                  "muted": null,
                  "autoplay": null,
                  "startAt": 0,
                  "endAt": 5,
                  "posterBlockId": null,
                  "fullsize": null,
                  "videoId": "045de1f71b8dc900cc57684a87d6c9c7",
                  "videoVariantLanguageId": null,
                  "source": "cloudflare",
                  "title": "test",
                  "description": null,
                  "image": "https://customer-209o3ptmsiaetvfx.cloudflarestream.com/045de1f71b8dc900cc57684a87d6c9c7/thumbnails/thumbnail.jpg?time=2s",
                  "duration": 5,
                  "objectFit": null,
                  "video": null,
                  "action": {
                      "__typename": "NavigateAction",
                      "parentBlockId": "92f5cd2e-3a0c-4aea-8a66-a5cf219f9d5d",
                      "gtmEventName": "NavigateAction"
                  },
                  "children": []
              }
```

Comment (storyworks) [reply libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.tsx:71]: Ah great, thanks. Are you able to update the comments on line 72 and line 82 as well then? Or remove if you think it's redundant?

**Review by storyworks — APPROVED** [review]

File: libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.spec.tsx [hunk 2]
```diff
@@ -147,12 +147,17 @@ describe('ContainedCover', () => {
     expect(posterImage).toHaveAttribute('aria-details', imageBlock.src)
   })
 
-  it('should render background video with video.image', () => {
+  it('should render background video with default cloudflare thumbnail image', () => {
```
Comment (storyworks) [inline libs/journeys/ui/src/components/Card/ContainedCover/ContainedCover.spec.tsx:150]: Hey, I think if you're going to do this, we now miss the use case where we render a background image with `videoBlock.video.image` which occurs with arclight videos that have a thumbnail image set on it by default.

**Review by storyworks — APPROVED** [review]

**Pull request merged**